benchmarks/sparse/dlmc/spmm.py

```diff
@@ -0,0 +1,150 @@
+                        default='matmul')
+    parser.add_argument('--spmm_type',
+                        type=str,
+                        help='sparse-sparse or sparse-dense',
+                        default='sparse-dense')
+    parser.add_argument('--output',
+                        type=str,
+                        help='dataframe output path',
+                        default='/tmp/matmul_bench.pkl')
+    parser.add_argument('--device',
+                        type=str,
+                        help='device (cpu or cuda)',
+                        default='cpu')
+
+    args = parser.parse_args()
+    print('path     =', args.path)
+    print('dataset       =', args.dataset)
+    print('spmm_type        =', args.spmm_type)
+    print('operation     =', args.operation)
+    print('output        =', args.output)
+    print('device        =', args.device)
+
+    dataset_path = args.path
+    dataset_name = args.dataset
+    dataset_path = f"{dataset_path}/{dataset_name}"
+    df_output_path = args.output
+    spmm_type = args.spmm_type
+    device = args.device
+
+    tasks = []
+    if args.operation == 'matmul':
+        if device == 'cpu':
+            tasks = [
+                ("matmul", device, "torch", "torch.mm(dx, dy)"),
+                ("matmul", device, "torch.sparse", "torch.sparse.mm(x, y)"),
+                ("matmul", device, "scipy", "scipy_coo_matmul(sx, sy)"),
+            ]
+        else:
+            tasks = [
+                ("matmul", device, "torch", "torch.mm(dx, dy)"),
+                ("matmul", device, "torch.sparse", "torch.sparse.mm(x, y)"),
+            ]
+    else:
+        tasks = [
+            ("backward", device, "torch", "torch_backward(dx, dy)"),
+            ("backward", device, "torch.sparse", "sparse_torch_backward(x, y)"),
+        ]
+    serialized_results = []
+    repeats = 2
+    timers = [
+        benchmark_utils.Timer(
+            stmt=stmt,
+            globals={
+                "scipy_coo_matmul": scipy_coo_matmul,
+                "torch_backward": torch_backward,
+                "sparse_torch_backward": sparse_torch_backward,
+                "sx": to_coo_scipy(x) if x.is_sparse and device == 'cpu' else x.numpy(),
+                "sy": to_coo_scipy(y) if y.is_sparse and device == 'cpu' else y.numpy(),
+                "x": x,
+                "y": y,
+                "dx": x.to_dense() if x.is_sparse else x,
+                "dy": y.to_dense() if y.is_sparse else y,
+            },
+            label=label,
+            sub_label=sub_label,
+            description=f"{sparsity}",
+            env=device,
+        ) for hidden_size in [512]
+        for sparsity in [0.5, 0.7, 0.8, 0.9, 0.95, 0.98]
+        for label, device, sub_label, stmt in tasks
+        for x, y in load_spmm_dataset(dataset_path, hidden_size, sparsity, spmm_type, device)
+    ]
+    measurements = []
+
+    for i, timer in enumerate(timers * repeats):
+        m = timer.blocked_autorange(min_run_time=0.05)
```

Comment:
Is this enough to get stable measurements?

Reply:
For the maximum problem size, `hidden_size=2048`,  is not enough. I changed it to 1 by default which should be sufficient to be stable and this now can be adjusted in the terminal. 
`    parser.add_argument('--timer_min_run_time', default=1, type=int)`